board.cpp

```diff
@@ -0,0 +1,83 @@
+#include "board.h"
+
+
+Board::Board() {
+	this->board = std::vector<int>(BOARD::TRUE_BOARD_SIZE, 0);
+	this->blocksPerRow = std::vector<int>(BOARD::COLUMN_SIZE, 0);
+	this->currentHeight = 1;
+	
+	// place buffer at beginning and end of each row
+	for (int row = 0, size = BOARD::TRUE_COLUMN_SIZE; row < size; ++row) {
```

Comment:
I know what you are going here b/c you explained it to me but its hard to figure it out just by reading this comment + code. A well named macro for -1 might help (OUT_OF_BOUNDS_SENTINEL?).